server/sockets/list.js

```diff
@@ -413,42 +433,62 @@ const changeListType = (socket, dashboardClients, cohortClients, listClients) =>
 
           if (dashboardClients.size > 0) {
             if (type === ListType.LIMITED && removedViewers) {
-              emitForMany(
-                removedViewers,
-                dashboardClients,
-                socket,
-                ListActionTypes.DELETE_SUCCESS,
-                listId
-              );
+              removedViewers.forEach(id => {
+                const userId = id.toString();
+
+                if (dashboardClients.has(userId)) {
+                  socket.broadcast
+                    .to(dashboardClients.get(userId))
+                    .emit(ListActionTypes.DELETE_SUCCESS, listId);
+                }
+              });
             }
 
-            emitListForMany(
-              viewersIds,
-              dashboardClients,
-              socket,
-              ListActionTypes.FETCH_META_DATA_SUCCESS,
-              list
-            );
+            viewersIds.forEach(id => {
+              const userId = id.toString();
+
+              if (dashboardClients.has(userId)) {
+                socket.broadcast
+                  .to(dashboardClients.get(userId))
+                  .emit(ListActionTypes.FETCH_META_DATA_SUCCESS, {
+                    [list._id]: { ...responseWithList(list, userId) }
+                  });
+              }
+            });
           }
 
           if (cohortClients.size > 0) {
             if (type === ListType.LIMITED && removedViewers) {
-              emitForMany(
-                removedViewers,
-                cohortClients,
-                socket,
-                ListActionTypes.DELETE_SUCCESS,
-                listId
-              );
+              removedViewers.forEach(id => {
+                const userId = id.toString();
+
+                if (cohortClients.has(userId)) {
+                  const { viewId, socketId } = cohortClients.get(userId);
+
+                  if (viewId === cohortId.toString()) {
+                    socket.broadcast
+                      .to(socketId)
+                      .emit(ListActionTypes.DELETE_SUCCESS, listId);
+                  }
+                }
+              });
```

Comment:
same here, please check if `cohortClients.size` is necessary if you put `cohortClients.has(userId)` in the first if

Reply:
I removed checking the size and refactored the code but a little bit different.